src/Jupyter/Extensions.cs

```diff
@@ -202,6 +202,18 @@ public static bool TryDecodeParameter<T>(this Dictionary<string, string> paramet
                 parameterValue = parameterValue.Substring(1, parameterValue.Length - 2);    // Strip off the enclosing quotes
                 parameterValue = parameterValue.Replace("\\\"", "\"");                      // Don't escape the interior quotes
             }
+
+            // This is a temporary fix in order to send JSON encoded data as strings to the Azure submission client
+            var intermediate = JsonConvert.DeserializeObject<ImmutableDictionary<string, object>>(parameterValue);
+            if (intermediate == null) {
```

Comment:
I'd suggest not using `==` to compare to null. The `is` operator is more reliable as it cannot be overriden by operator overloading.